.github/workflows/test.yml

```diff
@@ -0,0 +1,215 @@
+              -DBUILD_PY_FFI=ON \
+              -DBUILD_MULTI_GPU=ON \
+              -DCMAKE_CUDA_FLAGS="-lineinfo" \
+              -DUSE_NVTX=ON \
+              -DSM=80 \
+              -DCMAKE_CUDA_ARCHITECTURES=80 \
+              -DBUILD_TEST=OFF
+          make -j$(nproc) && make install
+      - name: Install lmdeploy
+        run: |
+          python3 -m pip install packaging protobuf transformers_stream_generator
+          python3 -m pip install -r requirements.txt -r requirements/test.txt
+          python3 -m pip install .
+      - name: Check env
+        run: |
+          python3 -m pip list
+          lmdeploy check_env
+      - name: Test lmdeploy
+        run: |
+          echo "TODO: awaiting PR of adding autotest"
+          # pytest autotest ${{github.event.inputs.markers}} --alluredir=allure-results --clean-alluredir
+      - name: Generate reports
+        if: always()
+        run: |
+          if test -D "allure-results"; then
+            export date_today="$(date +'%Y%m%d-%H%M%S')"
+            export report_dir="$REPORT_DIR/$date_today"
+            echo "Save report to $ALLURE_DIR"
+            allure generate -c -o $report_dir
+          fi
+      - name: Clear workfile
+        if: always()
+        run: |
+          export workdir=$(pwd)
+          cd ..
+          rm -rf $workdir
+          mkdir $workdir
+          chmod -R 777 $workdir
+
+  test_triton:
+    runs-on: [self-hosted, linux-a100]
+    timeout-minutes: 4320 # 72hours
+    environment: 'prod'
+    env:
+      HF_MODEL: /nvme/qa_test_models/internlm-chat-20b
+      WORKDIR: /nvme/qa_test_models/triton_workspace
+      TB_MODEL: internlm-chat-20b-fp16-tp2
+      GRPC_PORT: 33337
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+      - name: Create test container
+        run: |
+          export CONTAINER_ID=$(docker create \
+            --rm \
+            --gpus='"device=0,1"' \
+            --shm-size 16g \
+            --cap-add=SYS_PTRACE \
+            --cap-add=SYS_ADMIN \
+            --security-opt seccomp=unconfined \
+            --name lmdeploy-ci-triton \
+            --workdir /__w/lmdeploy/lmdeploy \
+            --env PIP_CACHE_DIR=/root/.cache/pip \
+            --env NCCL_LAUNCH_MODE=GROUP \
+            -v $(pwd)/../../:/__w \
+            -v ${HF_MODEL}:/root/workspace/hf_model \
+            -v ${WORKDIR}:/root/workspace/workdir \
+            -v ${HOST_PIP_CACHE_DIR}:/root/.cache/pip \
+            -v ${HOST_LOCALTIME}:/etc/localtime:ro \
+            openmmlab/lmdeploy:latest tail -f /dev/null \
+             )
+          docker start $CONTAINER_ID
+          echo "CONTAINER_ID=$CONTAINER_ID"
+          echo "CONTAINER_ID=$CONTAINER_ID"  >> $GITHUB_ENV
+      - name: Build lmdeploy from source
+        run: |
+          docker exec $CONTAINER_ID cp -r ../../build build
+          docker exec --workdir /__w/lmdeploy/lmdeploy/build \
+            --env http_proxy=${{secrets.PROXY}} \
```

Comment:
[test triton](https://github.com/InternLM/lmdeploy/actions/runs/7000380916/job/19040937839?pr=760#step:4:4) failed because the secrets.PROXY cannot be accessed in pull requeset before it's merged in to main branch. The job is tested ok on the forked repo: https://github.com/RunningLeon/lmdeploy/actions/runs/6999607225/job/19040778718#step:4:4